tensorboard/plugins/graph/polymer3/tf_graph_loader/tf-graph-dashboard-loader.ts

```diff
@@ -12,186 +12,207 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+import {getRouter} from '../../../../components_polymer3/tf_backend/router';
 
-          const shouldFetchGraph =
-            !this._graphRunTag ||
-            this._graphRunTag.run !== run ||
-            this._graphRunTag.tag !== requiredOpGraphTag;
-          const maybeFetchGraphPromise = shouldFetchGraph
-            ? this._load({
-                run,
-                tag: requiredOpGraphTag,
-                type: tf.graph.SelectionType.OP_GRAPH,
-              })
-            : Promise.resolve();
-          const params = new URLSearchParams();
-          params.set('tag', tag);
-          params.set('run', run);
-          const metadataPath = tf_backend
-            .getRouter()
-            .pluginRoute('graphs', '/run_metadata', params);
-          return maybeFetchGraphPromise.then(() =>
-            this._readAndParseMetadata(metadataPath)
-          );
-        }
-        default:
-          return Promise.reject(
-            new Error(`Unknown selection type: ${selectionType}`)
-          );
+interface GraphRunTag {
+  run: string;
+  tag?: string;
+}
+
+/**
+ * Data loader for tf-graph-dashboard.
+ *
+ * The loader loads op graph, conceptual graphs, and RunMetadata associated with
+ * an op graph which is the major difference from the tf-graph-loader which is
+ * only capable of loading an op graph. Another difference is that the loader
+ * takes `selection` from the tf-graph-controls as an input as opposed to URL
+ * path of an data endpoint.
+ */
+@customElement('tf-graph-dashboard-loader')
+class TfGraphDashboardLoader extends LegacyElementMixin(PolymerElement) {
+  @property({type: Array})
+  datasets: any[];
+  /**
+   * @type {{value: number, msg: string}}
+   *
+   * A number between 0 and 100 denoting the % of progress
+   * for the progress bar and the displayed message.
+   */
+  @property({
+    type: Object,
+    notify: true,
+  })
+  progress: object;
+  @property({type: Object})
+  selection: any;
+  /**
+   * TODO(stephanwlee): This should be changed to take in FileList or
+   * the prop should be changed to `fileInput`.
+   * @type {?Event}
+   */
+  @property({type: Object})
+  selectedFile: object;
+  @property({
+    type: Object,
+  })
+  compatibilityProvider = new tf_graph_op.TpuCompatibilityProvider();
+  @property({
+    type: Object,
+  })
+  hierarchyParams = tf_graph_hierarchy.DefaultHierarchyParams;
+  @property({
+    type: Object,
+    readOnly: true, //readonly so outsider can't change this via binding
+    notify: true,
+  })
+  outGraphHierarchy: object;
+  @property({
+    type: Object,
+    readOnly: true, //readonly so outsider can't change this via binding
+    notify: true,
+  })
+  outGraph: object;
+  @property({
+    type: Object,
+    readOnly: true, // This property produces data.
+    notify: true,
+  })
+  outStats: object;
+  @property({type: Object})
+  _graphRunTag: GraphRunTag;
+  _template = null;
```

Comment:
Hmm, can you make sure if this works in Polymer 3? Maybe `static readonly template = null` still works.

Reply:
Checked that it still works.